addons/gdUnit3/src/cmd/CmdArgumentParser.gd

```diff
@@ -0,0 +1,60 @@
+class_name CmdArgumentParser
+extends Reference
+
+var _options :CmdOptions
+var _tool_name :String
+var _parsed_commands :Dictionary = Dictionary()
+
+func _init(options :CmdOptions, tool_name :String):
+	_options = options
+	_tool_name = tool_name
+
+func parse(args :Array) -> Result:
+	_parsed_commands.clear()
+	
+	# parse until first program argument
+	while not args.empty():
+		var arg :String = args.pop_front()
+		if arg.find(_tool_name):
+			args.pop_front()
+			break
+	
+	if args.empty():
+		return Result.success(0)
+	
+	# now parse all arguments
+	while not args.empty():
+		var cmd :String = args.pop_front()
+		var option := _options.get_option(cmd)
+		
+		if option:
+			if _parse_cmd_arguments(option, args) == -1:
+				return Result.error("The '%s' command requires an argument!" % option.short_command())
+		else:
+			return Result.error("Unknown '%s' command!" % cmd)
+	return Result.success(commands().size())
```

Comment:
it should return the commands instead of size, than we can delete `func commands() -> Array:` and use the result value